ssl/t1_trce.c

```diff
@@ -683,6 +695,24 @@ static int ssl_print_extension(BIO *bio, int indent, int server,
         if (extlen != xlen + 2)
             return 0;
         return ssl_trace_list(bio, indent + 2, ext + 2, xlen, 2, ssl_groups_tbl);
+    case TLSEXT_TYPE_application_layer_protocol_negotiation:
+        if (extlen < 2)
+            return 0;
+        xlen = (ext[0] << 8) | ext[1];
+        if (extlen != xlen + 2)
+            return 0;
+        ext += 2;
```

Comment:
Maybe turn this into a for loop and put line 712 into the statement as well?  Yes doing that would mean plen decl moves up.  Just a suggestion.